src/Rules/RuleBuilder.php

```diff
@@ -39,6 +45,10 @@ public function addThat(Expression $that): self
 
     public function addShould(Expression $should): self
     {
+        if ($this->negateShoulds) {
+            $should = new NegateDecorator($should);
+        }
```

Comment:
Maybe I am too low on ☕  but I can't wrap my head around this, would you mind explaining it to me? 
If the should method in NoClass is defined as follow

```
$this->ruleBuilder->addShould(new NegateDecorator($expression));
```

why do we need to add another `new NegateDecorator` here?